src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -426,6 +400,14 @@ public boolean onPortNumber(int portNumber) {
             throw new BindTransportException("Failed to bind to [" + port + "]", lastException.get());
         }
 
+        if (!DEFAULT_PROFILE.equals(name)) {
+            InetSocketAddress boundAddress = (InetSocketAddress) serverChannels.get(name).getLocalAddress();
+            int publishPort = settings.getAsInt("publish_port", boundAddress.getPort());
+            String publishHost = settings.get("publish_host", this.settings.get("transport.publish_host", this.settings.get("transport.host")));
+            InetSocketAddress publishAddress = createPublishAddress(publishHost, publishPort);
+            profileBoundAddresses.put(name, new BoundTransportAddress(new InetSocketTransportAddress(boundAddress), new InetSocketTransportAddress(publishAddress)));
```

Comment:
Ok, then for the profile settings you only read the publish_port and publish_host instead of the other settings. I see the difference now. Before it seemd like you wanted to read exactly the same settings, then I wondered why that part couldn't be shared.